### DISCUSSION TITLE
Compiler ESM-ESP

### DISCUSSION THREAD

**Discussion by isolhara** [emsesp/EMS-ESP32#1028] [General]
I have a Bosch fresh water station FF20. I decrypted the telegrams. Would like to integrate this into EMS-ESP. Compiling with Visual Code brings error. How can I compile the EMS-ESP to get a BIN file.

**Comment by proddy** [comment]
what fails? please post a screenshot

https://emsesp.github.io/docs/Coding/

**Comment by isolhara** [comment]
Unfortunately I don't know English and translated with Google. Would German also be possible? Attached is the screenshot
![Screenshot 2023-02-11 071432](https://user-images.githubusercontent.com/30491957/218244567-a8bcd767-bfc4-4750-9d88-72b300a83d54.png)

**Comment by proddy** [comment]
have you installed npm (nodejs)?

**Comment by isolhara** [comment]
No, that was not installed. Now it worked. Thanks very much.
![Screenshot 2023-02-11 101653](https://user-images.githubusercontent.com/30491957/218250678-465c1216-6ad8-42aa-9739-bfc00f0a09ae.png)

**Comment by proddy** [comment]
Ok great. I've also updated the documentation to mention the pre-requisite for nodejs. Thanks for pointing that out!
